apps/mobile-mzima-client/src/app/profile/information/components/profile-photo/profile-photo.component.ts

```diff
@@ -1,21 +1,135 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
-import { AlertService } from '@services';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { AlertService, SessionService } from '@services';
+import { map } from 'rxjs';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { MediaService } from 'libs/sdk/src/lib/services';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { UsersService } from 'libs/sdk/src/lib/services';
+import { UserInterface } from '@mzima-client/sdk';
 
+@UntilDestroy()
 @Component({
   selector: 'app-profile-photo',
   templateUrl: './profile-photo.component.html',
   styleUrls: ['./profile-photo.component.scss'],
 })
 export class ProfilePhotoComponent {
   @Input() photo: string;
-  @Output() deletePhoto = new EventEmitter();
-  @Output() changePhoto = new EventEmitter();
+  userId: string;
+  @Output() photoUpdated = new EventEmitter<boolean>();
 
-  constructor(private alertService: AlertService) {}
+  public currentUser: UserInterface;
+  constructor(
+    private alertService: AlertService,
+    private sessionService: SessionService,
+    private mediaService: MediaService,
+    private usersService: UsersService,
+  ) {}
 
-  public uploadNewPhoto(): void {
-    console.log('uploadNewPhoto');
-    this.changePhoto.emit();
+  selectPhoto(event: Event): void {
+    const input = event.target as HTMLInputElement;
+    const file = input?.files?.[0];
+    if (file) {
+      this.changePhoto(file);
+    }
+  }
+
+  changePhoto(file: File): void {
+    const reader = new FileReader();
+    console.log(reader);
+    console.log(file);
+    reader.onload = () => {
+      this.photo = reader.result as string;
+
+      this.mediaService.uploadFile(file, 'Test caption').subscribe((response: any) => {
```

Comment:
Should probably change 'Test Caption'!

Reply:
Got it. Should we name it "Profile Picture" instead?

Reply:
Maybe the user's display name?

Reply:
Yeah, I think this may work